source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,296 @@
+
+   * - Passphrase
+     - The password to decrypt the client private key if it is encrypted.
+
+.. tip::
+   
+   To learn more about the PEM format, see the Wikipedia entry on
+   :wikipedia:`Privacy-Enhanced Mail <Privacy-Enhanced_Mail>`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions``
+object so that the server can validate them before the client connects.
+You can reference your certificates in the following ways:
+
+- Create a ``SecureContext`` object to store certificates *(Recommended)*
+- Provide filepath strings that point to your certificates
+- Create ``Buffer`` objects to store certificates
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Store Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to configure
+your TLS connection. ``SecureContext`` objects are native to Node.js
+and allow you to keep all your TLS options in a single reusable object.
+
+To create a ``SecureContext`` object, you must first install the ``tls``
+package and use it in your application. Next, call the ``tls.createSecureContext()``
+method and pass the contents of your certificates in the options parameter.
+This method returns a ``SecureContext`` object that you can set as an option 
+in your ``MongoClientOptions`` object.
+
+The following code shows how to create a ``SecureContext`` object by
+calling the ``tls.createSecureContext()`` method and set the
+``secureContext`` client option:
+
+.. code-block:: js
+   :emphasize-lines: 2-6, 10
+   
+   // Create a SecureContext object
+   var sc = tls.createSecureContext({
+     ca: fs.readFileSync(`<path to CA certificate>`),
+     cert: fs.readFileSync(`<path to client certificate>`),
+     key: fs.readFileSync(`<path to client certificate>`),
+   });
+   
+   // Pass the SecureContext as a client option
+   const client = new MongoClient(uri, {
+     secureContext: sc,
+   });
+
+To learn more about the ``tls.createSecureContext()`` method and the
+``tls`` package, see the `Node.js TLS API documentation
+<https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions>`__.
+
+For a runnable example that uses the ``secureContext`` client option, see
+the :ref:`SecureContext Example <node-securecontext-full-example>`.
+
+Provide Certificate Filepaths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can include the filepaths for your certificates as client options to
+retrieve your certificates while connecting with TLS.
+
+The following code shows how to specify the ``tlsCAFile`` and
+``tlsCertificateKeyFile`` client options with certificate filepaths: 
+
+.. code-block:: js
+   :emphasize-lines: 3-4
+   
+   // Pass filepaths to client options
+   const client = new MongoClient(uri, {
+     tlsCAFile: `<path to CA certificate>`,
+     tlsCertificateKeyFile: `<path to client certificate>`,
+   });
```

Comment:
```suggestion
   const client = new MongoClient(uri, {
     tls: true,
     tlsCAFile: `<path to CA certificate>`,
     tlsCertificateFile: `<path to public client certificate>`,
     tlsCertificateKeyFile: `<path to private client key>`,
   });
```

Missing `tlsCertificateFile` and even though it could be set in the uri, just to make sure, we should have all the examples include tls:true, it it still required to enable tls even with the file paths provided.